photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/search/RandomSearch.scala

```diff
@@ -53,17 +51,29 @@ class RandomSearch[T](
    * Searches and returns n points in the space with the given prior observations
    *
    * @param n the number of points to find
-   * @param observations observations made prior to searching, as (paramVector, evaluationValue) tuples
+   * @param observations observations made prior to searching, as (paramVector, evaluationValue) tuples in the current
+   *                     dataset
+   * @param priorObservations observations from the past datasets. These are consider as mean centered.
    * @return the found points
    */
-  def findWithPrior(n: Int, observations: Seq[(DenseVector[Double], Double)]): Seq[T] = {
+  def findWithPrior(
+      n: Int,
+      observations: Seq[(DenseVector[Double], Double)],
+      priorObservations: Option[Seq[(DenseVector[Double], Double)]] = None): Seq[T] = {
     require(n > 0, "The number of results must be greater than zero.")
 
     // Load the initial observations
     observations.init.foreach { case (candidate, value) =>
       onObservation(candidate, value)
     }
 
+    if(priorObservations.isDefined) {
+      // Load the prior observations. We add all of them since we do not iterate over these.
+      priorObservations.get.foreach { case (candidate, value) =>
+        onObservation(candidate, value, priorObservations.isDefined)
+      }
+    }
```

Comment:
This is another case where pattern matching would be more idiomatic:

```scala
priorObservations match {
  case Some(priorObs) => priorObs.foreach { case (candidate, value) =>
    onObservation(candidate, value, priorObservations.isDefined)
  }
}
```

Also, adding an `onPriorObservation` function as suggested above would eliminate the need for the `isDefined`.

Reply:
Thanks. This is now changed.